src/widgets/list.rs

```diff
@@ -159,23 +515,76 @@ impl<'a> List<'a> {
+    /// when this spacing is allocated.
+    ///
+    /// - [`HighlightSpacing::Always`] will always allocate the spacing, regardless of whether an
+    ///   item is selected or not. This means that the table will never change size, regardless of
+    ///   if an item is selected or not.
+    /// - [`HighlightSpacing::WhenSelected`] will only allocate the spacing if an itemis selected.
+    ///   This means that the table will shift when an item is selected. This is the default setting
+    ///   for backwards compatibility, but it is recommended to use `HighlightSpacing::Always` for a
+    ///   better user experience.
+    /// - [`HighlightSpacing::Never`] will never allocate the spacing, regardless of whether an item
+    ///   is selected or not. This means that the highlight symbol will never be drawn.
+    ///
+    /// This is a fluent setter method which must be chained or used as it consumes self
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// # let items = vec![ListItem::new("Item 1")];
+    /// let list = List::new(items).highlight_spacing(HighlightSpacing::Always);
+    /// ```
     #[must_use = "method moves the value of self and returns the modified value"]
     pub fn highlight_spacing(mut self, value: HighlightSpacing) -> Self {
         self.highlight_spacing = value;
         self
     }
 
+    /// Defines the list direction (up or down)
+    ///
+    /// Defines if the `List` is displayed *top to bottom* (default) or *bottom to top*. Use
+    /// [`Corner::BottomLeft`] to go *bottom to top*. **Any** other variant will go *top to bottom*.
+    ///
+    /// This is set to [`Corner::TopLeft`] by default.
+    ///
+    /// This is a fluent setter method which must be chained or used as it consumes self
+    ///
+    /// ## Note
+    ///
+    /// Despite its name, this method doesn't change the horizontal alignment, i.e. the `List`
+    /// **won't** start in a corner.
+    ///
+    /// # Example
+    ///
+    /// Same as default, i.e. *top to bottom*. Despite the name implying otherwise.
+    /// ```rust
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// # let items = vec![ListItem::new("Item 1")];
+    /// let list = List::new(items).start_corner(Corner::BottomRight);
+    /// ```
+    ///
+    /// Bottom to top
+    /// ```rust
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// # let items = vec![ListItem::new("Item 1")];
+    /// let list = List::new(items).start_corner(Corner::BottomLeft);
+    /// ```
     #[must_use = "method moves the value of self and returns the modified value"]
     pub fn start_corner(mut self, corner: Corner) -> List<'a> {
         self.start_corner = corner;
         self
     }
 
+    /// Returns the number of [`ListItem`]s in the list
     pub fn len(&self) -> usize {
         self.items.len()
     }
 
+    /// Tells if the `List` is empty
```

Comment:
We can use maybe something more explicit such as "Returns "true" if the list is empty" - it also aligns with the Rust stdlib docs

Reply:
Good idea, I literally copied the `Vec` documentation